apis/python/src/tiledbsoma/dataframe.py

```diff
@@ -258,7 +258,8 @@ def read(
                             )
                         sr.set_dim_points(dim_name, dim_ids)
                     elif isinstance(dim_ids, slice):
-                        lo_hi = util.slice_to_range(dim_ids)
+                        ned = A.nonempty_domain()
+                        lo_hi = util.slice_to_range(dim_ids, ned[i]) if ned else None
```

Comment:
Not sure if you have another commit coming -- please pardon if these remarks are premature :)

Let's add a comment saying that non-empty domain is None precisely when there is no data in the array at all